gamemodes/terrortown/gamemode/shared/sh_shop.lua

```diff
@@ -0,0 +1,320 @@
+---
+-- A class to handle all shop requirements
+-- @author ZenBre4ker
+-- @class Shop
+
+shop = shop or {}
+
+shop.statusCode = {
+	SUCCESS = 1,
+	SUCCESSIGNORECOST = 2,
+	INVALIDPLAYER = 3,
+	INVALIDID = 4,
+	NOTEXISTING = 5,
+	NOTENOUGHCREDITS = 6,
+	NOTBUYABLE = 7,
+	NOTENOUGHPLAYERS = 8,
+	LIMITEDBOUGHT = 9,
+	NOTBUYABLEFORROLE = 10,
+	PENDINGORDER = 11,
+	NOSHOP = 12,
+	NORANDOMSHOP = 13,
+	NOTINSHOP = 14,
+	BLOCKEDBYOLDHOOK = 15,
+	BLOCKEDBYTTT2HOOK = 16,
+}
+
+---
+-- Check if an equipment is currently buyable for a player
+-- @param Player ply The player to buy the equipment for
+-- @param string equipmentId The name of the equipment to buy
+-- @return bool True, if equipment can be bought
+-- @return number The shop.statusCode, that lead to the decision
+-- @realm shared
+function shop.IsEquipmentBuyableFor(ply, equipmentId)
```

Comment:
```suggestion
function shop.CanBuyEquipment(ply, equipmentId)
```
just a suggestions, but how about keeping this in line with `RerollShop` and `CanRerollShop`?